### PR TITLE
object: Support numerical matchers in search queries

### PR DESCRIPTION
**Pull request by cthulhu-rider**
* based on and blocked by https://github.com/nspcc-dev/neofs-api/pull/285

### CODE REVIEW

File: object/search.go [hunk 1]
```diff
@@ -86,15 +121,27 @@ func (m SearchMatchType) String() string {
...
-	ok := g.FromString(s)
-
-	if ok {
-		*m = SearchMatchFromV2(g)
+	switch s {
+	default:
+		var g v2object.MatchType
+
+		ok := g.FromString(s)
+
```
Comment (carpawell) [inline object/search.go:129]: required empty line?

File: object/search.go [hunk 2]
```diff
@@ -200,6 +247,8 @@ func (f *SearchFilters) addFilter(op SearchMatchType, key string, val stringEnco
 }
 
 // AddFilter adds a filter to group by simple plain parameters.
+//
+// If op is numerical (e.g. [MatchNumGT], value must be a base-10 integer.
```
Comment (carpawell) [inline object/search.go:251]:
```suggestion
// If op is numerical (e.g. [MatchNumGT]), value must be a base-10 integer.
```

**Review by carpawell — APPROVED** [review]

File: object/search.go [hunk 3]
```diff
@@ -37,6 +41,14 @@ func (m SearchMatchType) ToV2() v2object.MatchType {
 		return v2object.MatchNotPresent
 	case MatchCommonPrefix:
 		return v2object.MatchCommonPrefix
+	case MatchNumGT:
+		return 5
```
Comment (carpawell) [inline object/search.go:45]: looks extremely strange to me but since you are _that_ against api-go, i cannot say anything about it

Comment (roman-khimov) [reply object/search.go:45]: This line sucks, really.

Comment (cthulhu-rider) [reply object/search.go:45]: i can add one more PR if u need

Comment (carpawell) [reply object/search.go:45]: at least constants from `.pb` file. can be merged in the https://github.com/nspcc-dev/neofs-api-go/pull/437. or we can add the same `v2object` constants, not a big work

File: object/search.go (hunk 1 above)
Comment (cthulhu-rider) [reply object/search.go]: untouched now

File: object/search.go (hunk 3 above)
Comment (cthulhu-rider) [reply object/search.go]: done

**Review by carpawell — APPROVED** [review]
Can be rebased and linter will be happy then.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**